docs/GraphRepresentation.md

```diff
@@ -0,0 +1,321 @@
+
+# Graph representation
+
+It is helpful to discuss candidate lists, relevance scores and treatments
+in terms of an __add-on relational graph__ structure,
+as much of the intuition behind the treatments
+is drawn from this representation.
+
+An add-on relational graph is a directed graph $G = (V,E)$ in which:
+
+- each vertex is an add-on appearing in the dataset
+- there is an edge from A to B if add-on B is considered _related_ to add-on A
+- each edge has an associated weight
+    indicating the strength of the relationship.
+
+The data used in generating recommendations can be summarized
+by such a graph, where
+an add-on's candidate list corresponds to its set of neighbours in the graph,
+and relevance scores are given by the edge weights.
+In this setting:
+
+- A relational graph is induced directly by the coinstallation data,
+    considering add-ons related if they are coinstalled,
+    and using coinstallation counts as weights.
+- Treatments can be thought of as transformations
+    which modify add-on relational graphs by
+    adding or deleting edges, or adjusting the edge weights.
+- The final selection of recommendations can itself be viewed as a treatment,
```

Comment:
Since we studied recommendations (looking for cycles) in terms of their graph representation, I thought this tied everything together nicely. Getting from the treated graph to the recommendation graph is also a graph transformation, and I felt like keeping that transformation isolated from the other "treatment" transformations made things less clear. I'm open to suggestions, though.